corehq/apps/hqwebapp/views.py

```diff
@@ -1384,6 +1394,7 @@ def post(self, request, *args, **kwargs):
         return HttpResponseRedirect(reverse('oauth2_provider:detail', args=[str(base_application.id)]))
 
 
+@csrf_exempt
```

Comment:
I can't tell, but is `check_sso_login_status` expected to be called by us, or the third-party SSO site? Given that it ends up suggesting the user navigate to the IDP's site, I'm assuming the former...in which case, are we sure `@csrf_exempt` is correct here?

Reply:
this is absolutely needed! I spent hours trying to find a workaround for this, but ultimately if you have two tabs open, re-login on one of them...the other one now needs a CSRF token to get a response back from this view because there is an active session at that moment. As a result the view 403s when trying to see if the user should login with SSO or a password. The result is that it defaults to a password, even for an SSO user, and that is absolutely not the experience we want.

Reply:
My concern was what security holes this could open. From a brief glance, it doesn't seem like a malicious attacker could do anything by sending a request to this endpoint, right? Seems like it should be fine to leave it CSRF-exempt since it doesn't allow any modification of data or expose user information.

Reply:
yes, I evaluated safety based on the fact there is nothing you can do with the information gathered. You only know what the login URL for an Identity Provider might be (which can be guessed anyway). At that point, you still need access to the Identity Provider to do anything. So...it's 100% safe.